lib/count_words.rb

```diff
@@ -0,0 +1,18 @@
+class CountWords
+  attr_reader :counts
+
+  def initialize
+    clear_counts
+  end
+
+  def clear_counts
+    @counts = Hash.new(0)
+  end
+
+  def count_words(input_string)
+    words = input_string.scan(/'?[a-zA-Z]+(?:[-'][a-zA-Z]+)*/).map(&:downcase)
```

Comment:
You could use the block form of `scan` à la:

```ruby
input_string.scan(WORD_PATTERN) do |word|
  @counts[word.downcase] += 1
end
```
A snarky tweeter might tweet: "Local variables are a smell."